common/config_schema.go

```diff
@@ -31,6 +31,15 @@ type SchemaObject struct {
 	Requirements []RequiredFields `json:"requirements" msgpack:"requirements"`
 }
 
+type RecordKey = struct {
+	Name         string         `json:"name" msgpack:"name"`
+	Label        Label          `json:"label,omitempty" msgpack:"label,omitempty"` // Human readable label.
+	Description  string         `json:"description" msgpack:"description"`
+	DataType     SchemaDataType `json:"data_type" msgpack:"data_type"`
+	DisplayIndex int            `json:"display_index,omitempty" msgpack:"display_index,omitempty"`
+	PlaceHolder  string         `json:"placeholder" msgpack:"placeholder"` // Placeholder to display for this field.
+}
+
```

Comment:
A lot of this seems redundant, like DisplayIndex? How is it different than the KeyDisplayIndex?

Reply:
I thought it would read clearer to have a field for "key" and a field for "fields" as siblings that follow the SchemaElement structure.

But while I was intending for this to be a regular SchemaElement so that I wouldn't need to create a new type for it, I realized that I need to be able to fit another field in there so I created a new type for this. 

Maybe there's a cleaner way to do this?

Reply:
oh sorry, important context here might be that I'll be getting rid of all the `key_<field>` stuff after I migrate reliable-tasking to the new ui. So that all that data is managed in this new key object